torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -152,26 +153,31 @@ void FusionExecutor::compileFusion(Fusion* fusion, CompileOptions options) {
 namespace {
 
 at::Tensor inferAndAlloc(
-    const TensorView* tv,
-    StatefulExpressionEvaluator& see,
+    const kir::TensorView* tv,
+    kir::ExpressionEvaluator& expr_eval,
     const CompileOptions& options,
     bool zero_init = false) {
   FUSER_PERF_SCOPE("inferAndAlloc");
 
   std::vector<int64_t> sizes;
-  for (auto id : TensorDomain::noReductions(tv->getMaybeRFactorDomain())) {
-    auto inferred_val = see.inferValue(id->rawExtent());
+
+  const auto domain = tv->domain();
+  const auto maybe_rfactor_domain =
+      domain->hasRFactor() ? domain->rfactorDomain() : domain->rootDomain();
+
+  for (auto id : kir::TensorDomain::noReductions(maybe_rfactor_domain)) {
+    const auto inferred_val = expr_eval.evaluate(id->rawExtent());
```

Comment:
Is this change because kir::TensorView doesn't have `getMaybeRFactorDomain()`? Can the function be added?

Reply:
yes, 'maybe_rfactor_domain` avoids the need for 'getMaybeRFactorDomain`.

I'd rather not add that method since it's an anti-pattern (a shallow module - an extra interface which hides very little logic). For example, I noticed places where the use of `getMaybeRFactorDomain` was questionable, which was not obvious w/o reviewing the code carefully.

Reply:
I see your point, but things like `TensorView::getRootDomain()`, `TensorView::getMaybeRFactorDomain()` are very common operations, so having shortcuts like these would also help avoid repetitive code.

Either way, what I'd like to see is the (eventual) consistency between the Fusion and Kernel IRs. If we decide not to have, e.g., `getMaybeRFactorDomain` in `kir::TensorView`, then we should also remove it in the Fusion IR `TensorView` as well. This could be done in a future PR.

Reply:
There were quite a few instances of this where I also agree. I'm pretty open for what we bring where, but consistency seems like something valuable to try and approach.